VersionComparisonClass.py

```diff
@@ -0,0 +1,91 @@
+import semantic_version as sv
+import json
+
+class CheckVersion():
+  def __init__(self, VersionRangeList, VersionToCheck):
+    """
+    Inputs 
+    - a list of versions in str format from the file generated from generate_SNYK_vulnerabilities
+      - Example: ["[3.2-alpha,3.3-beta-2)"] 
+      - Splits up by the ','. Hence, every range exists in the following format
+        - ['[3.2-alpha', '3.3-beta-2)']
+    - Version to check in the intervals
+    """
+    self.VersionRangeInput = [interval.strip() for versionRange in VersionRangeList for interval in versionRange.split(',')]
+    self.VersionToCheck = sv.Version.coerce(VersionToCheck)
+
+  def CheckVersionInRange(self):
+    """
+    Iterates through the list in self.VersionRangeInput. 
+    Treats the beginning and end of range as seperate versions and constructs Spec object.
+    Checks the given range in the Spec object. Returns True, if found.
+    Else, goes through the next set of ranges.
+    """
+    VersionRangeList = []
+    for versionPoint in self.VersionRangeInput:
+      VersionRangeList.append(self.UnpackRange(versionPoint))
+      if len(VersionRangeList) == 2:
```

Comment:
Why do you check for length 2? What if there is only 1 version affected by a vulnerability?

Reply:
Thanks for pointing that out. Check ea3690b.